src/lib_info.ml

```diff
@@ -121,6 +121,21 @@ let of_library_stanza ~dir ~ext_lib (conf : Dune_file.Library.t) =
     , List.map (conf.c_names @ conf.cxx_names) ~f:snd
     )
   in
+  let foreign_archives =
+    match conf.stdlib with
+    | Some { exit_module = Some m; _ } ->
+      let obj_name = Path.relative dir (Module.Name.uncapitalize m) in
+      { Mode.Dict.
+        byte =
+          Path.extend_basename obj_name ~suffix:".cmo" ::
+          foreign_archives.byte
+      ; native =
+          Path.extend_basename obj_name ~suffix:".cmx" ::
+          Path.extend_basename obj_name ~suffix:ext_obj ::
```

Comment:
How would that work?

Reply:
We have the function

```
val foreign_objects : t -> ext:string -> Path.t list
```

That only takes `~ext:ext_obj` when it's called.

So when we initialize it in Lib_info:

```
  let (foreign_archives, foreign_objects) =
    let stubs =
      if Dune_file.Library.has_stubs conf then
        [Dune_file.Library.stubs_archive conf ~dir ~ext_lib]
      else
        []
    in
    ({ Mode.Dict.
       byte   = stubs
     ; native =
         Path.relative dir (Lib_name.Local.to_string lib_name ^ ext_lib)
         :: stubs
     }
    , List.map (conf.c_names @ conf.cxx_names) ~f:snd
    )
  in
```

We can simply add the extension right here since you're already passing it.

Reply:
That makes sense, I did this